core/src/aztec/AZDetector.cpp

```diff
@@ -291,7 +292,21 @@ static int ModeMessage(const BitMatrix& image, const PerspectiveTransform& mod2P
 		words[i] = narrow_cast<int>(bits & 0xF);
 		bits >>= 4;
 	}
-	if (!ReedSolomonDecode(GenericGF::AztecParam(), words, numECCodewords))
+
+	bool decodeResult = ReedSolomonDecode(GenericGF::AztecParam(), words, numECCodewords);
+
+	if ((!decodeResult) && compact) {
+		// Is this a Rune?
+		for (int i = numCodewords - 1; i >= 0; --i) {
+			words[i] ^= 0b1010;
+		}
+		decodeResult = ReedSolomonDecode(GenericGF::AztecParam(), words, numECCodewords);
+
+		if (decodeResult)
```

Comment:
The `if` here tells us that it definitely _is_ a rune, otherwise it's just a failure to decode the core symbol. While that doesn't matter right now – we check it again and `return -1` right afterward – it would be better not to set isRune = true if that's not actually necessarily the case.

Like, imagine you're debugging a symbol that fails to decode by setting a breakpoint here, and you see that `isRune` is set to true, you would probably be misled into thinking that the code is being detected as a rune for some reason - but it's not.

Unless you want to stick the `return -1` in the if block? Like

```cpp
// line 310
		if (!decodeResult)
			return -1;

		isRune = true;
	}
```

though I think I'd like to make it explicit that `isRune` being set is directly caused by `decodeResult` being true.

```cpp
// line 310
		if (decodeResult)
			isRune = true;
		else
			return -1; // Failed to decode as either normal or rune
	}
```

Reply:
Fair enough. I respect the effort of making a sound technical argument. :)